zcash_client_backend/src/data_api/chain.rs

```diff
@@ -209,12 +213,45 @@ pub trait BlockSource {
         &self,
         from_height: Option<BlockHeight>,
         limit: Option<usize>,
-        with_row: F,
-    ) -> Result<(), error::Error<WalletErrT, Self::Error>>
+        with_block: F,
+    ) -> Result<Option<Range<BlockHeight>>, error::Error<WalletErrT, Self::Error>>
     where
         F: FnMut(CompactBlock) -> Result<(), error::Error<WalletErrT, Self::Error>>;
 }
 
+#[derive(Debug)]
+pub struct ScanSummary {
```

Comment:
- Needs documentation (both the struct, and its methods).
- Needs to be mentioned as an addition in the changelog.